### DISCUSSION TITLE
Rancher Desktop Design - Supporting Utilities

### DISCUSSION THREAD

**Discussion by rak-phillip** [rancher-sandbox/rancher-desktop#693] [Archive]
## Supporting Utilities

* Provide an area to control anything that might be considered a "Supporting Utility", like defining symlinks for macOS
* Add context to what we might be changing and how it will affect the environment through descriptions
* Explicitly commit changes through Apply actions

![4 - Support Utils](https://user-images.githubusercontent.com/835961/134973325-f7d4d957-3682-4ba2-a9e7-af381f297964.png)

> Using the space to share this initial design proposal from the UI group. I've add some quick notes for each section of Rancher Desktop that might change. If you have any questions about anything you see below, please ask and I'll provide some more detail. 
>
> Original Discussion #683
>
> Feedback is highly valued and welcome.

**Comment by jandubois** [comment]
Assuming we have [separate settings for each runtime](https://github.com/rancher-sandbox/rancher-desktop/discussions/683#discussioncomment-1393134) I think we should move each "Supporting Utility" to the settings for the runtime it belongs to, e.g.

* Kubernetes: helm, kubectl, kim
* Docker: docker-cli
* Containerd: nerdctl
* Podman: podman

Even though the implementation is the same (we create a symlink), the user context should be: "What can I configure for the runtime I'm about to use?".

**Reply by rak-phillip** [reply]
Perhaps, it might be a good idea to discuss what the future state might look like concerning support for multiple runtimes. Getting a better idea of how we can expect multiple runtimes to behave and how we might want to interact with each of them will help us make the right decision here.

**Comment by jandubois** [comment]
Maybe some utilities should appear on multiple settings pages:

If you configure Kubernetes with `containerd`, then `nerdctl` should also be included on the Kubernetes settings page (in addition to the Containerd settings page, if that one is enabled as well).

Similarly, if you configure Kubernetes with `moby`, then `docker` (cli) should also be included in the Kubernetes settings page.

Actually, so far these are the only utilities that can appear on multiple settings pages.

**Comment by mook-as** [comment]
Looking around, it looks like modern HIG (possibly due to phone / Apple influence) is all about instant apply:
- [macOS](https://developer.apple.com/design/human-interface-guidelines/macos/app-architecture/preferences/)
- [Gnome](https://wiki.gnome.org/Design/HIG/Planning/Configuration#Instant_Apply)
- [Windows](https://docs.microsoft.com/en-us/windows/apps/design/app-settings/guidelines-for-app-settings#general-recommendations)

I couldn't find specific wording for KDE, but I think the default there is still a separate apply button.

**Reply by rak-phillip** [reply]
Great catch! Thanks for calling this out. We'll see about getting this updated.
